src/otx/v2/adapters/torch/mmengine/mmaction/engine.py

```diff
@@ -0,0 +1,293 @@
+            pipeline = get_default_pipeline(subset="predict")
+        config = Config({})
+        if isinstance(model, torch.nn.Module) and hasattr(model, "_config"):
+            config = model._config  # noqa: SLF001
+        elif isinstance(model, dict) and "_config" in model:
+            config = model["_config"]
+        config["test_dataloader"] = {"dataset": {"pipeline": pipeline}}
+        if isinstance(model, dict):
+            model.setdefault("_config", config)
+        elif isinstance(model, torch.nn.Module):
+            model._config = config  # noqa: SLF001
+
+        if isinstance(checkpoint, Path):
+            checkpoint = str(checkpoint)
+        if task is not None and task != "VideoRecognition":
+            raise NotImplementedError
+
+        inferencer = MMAction2Inferencer(
+            rec=config,
+            rec_weights=checkpoint,
+            device=device,
+            input_format="rawframes",
+        )
+
+        return inferencer(img, batch_size=batch_size, **kwargs)
+
+    def export(
+        self,
+        model: torch.nn.Module | (str | Config) | None = None,
+        checkpoint: str | Path | None = None,
+        precision: str | None = "float32",  # ["float16", "fp16", "float32", "fp32"]
+        task: str | None = "VideoRecognition",
+        codebase: str | None = "mmaction",
+        export_type: str = "OPENVINO",  # "ONNX" or "OPENVINO"
+        deploy_config: str | dict | None = None,
+        device: str = "cpu",
+        input_shape: tuple[int, int] | None = None,
+    ) -> dict:
+        """Export a PyTorch model to a specified format for deployment.
+
+        Args:
+            model (Optional[Union[torch.nn.Module, str, Config]]): The PyTorch model to export.
+            checkpoint (Optional[Union[str, Path]]): The path to the checkpoint file to use for exporting.
+            precision (Optional[str]): The precision to use for exporting.
+                Can be one of ["float16", "fp16", "float32", "fp32"].
+            task (Optional[str]): The task for which the model is being exported. Defaults to "VideoRecognition".
+            codebase (Optional[str]): The codebase for the model being exported. Defaults to "mmaction".
+            export_type (str): The type of export to perform. Can be one of "ONNX" or "OPENVINO". Defaults to "OPENVINO"
+            deploy_config (Optional[str]): The path to the deployment configuration file to use for exporting.
+                File path only.
+            device (str): The device to use for exporting. Defaults to "cpu".
+            input_shape (Optional[Tuple[int, int]]): The input shape of the model being exported.
+            **kwargs: Additional keyword arguments to pass to the export function.
+
+        Returns:
+            dict: A dictionary containing information about the exported model.
+        """
+        if not AVAILABLE:
+            msg = "MMXEngine's export is dependent on mmdeploy."
+            raise ModuleNotFoundError(msg)
```

Comment:
This part is what I mentioned.
```suggestion
        if not IS_MMDEPLOY_AVAILABLE:
            msg = "MMXEngine's export is dependent on mmdeploy."
            raise ModuleNotFoundError(msg)
```

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2627/commits/428a39c498376a7852400c5618d8c9d6cca72810